frontend/src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,15 @@
+import { createContext } from "react";
+
+export const AuthContext = createContext(null);
+const user = {
+  name: "Bobb",
+};
+export const AuthProvider = ({ children }) => {
+  const checkAuth = token => {
+    if (!token) {
+      return "/auth/login";
+    }
+    return null;
+  };
+  return <AuthContext.Provider value={{ checkAuth, user }}>{children}</AuthContext.Provider>;
+};
```

Comment:
This is no good. 
The user should be stored in a state variable like `const [user, setUser] = useState(null)` which then gets passed down into context provider value. Also, the check function, is likely not going to receive any argument, rather, it should refetch data from backend and update the state variable accordingly.
```js
function checkAuth() {
const  request = fetch() // fetch request to API
  if (request.success) {
    setUser(request.user)
  }
  //...
}
```